src/core/task/Task.ts

```diff
@@ -2335,9 +2335,16 @@ export class Task extends EventEmitter<TaskEvents> implements TaskLike {
 						})
 					}
 
+					// Check if we should preserve reasoning in the assistant message
+					let finalAssistantMessage = assistantMessage
+					if (reasoningMessage && this.api.getModel().info.preserveReasoning) {
+						// Prepend reasoning in XML tags to the assistant message so it's included in API history
+						finalAssistantMessage = `<thinking>${reasoningMessage}</thinking>\n${assistantMessage}`
```

Comment:
For MiniMax in particular I think they use the <think> tag. Maybe worth confirming with them exactly what the right format is to make things work better?